### ISSUE TITLE
crashes if maze size is 0

### ISSUE THREAD

**Issue by matthiaskrgr** [tbelaire/maze_w25#1]
````
Enter a size of maze:0
Enter the number of trolls:0
thread 'main' panicked at 'index out of bounds: the len is 1 but the index is 1', /checkout/src/libcore/slice/mod.rs:2052:14
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:211
   3: std::panicking::default_hook
             at libstd/panicking.rs:227
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:477
   5: std::panicking::continue_panic_fmt
             at libstd/panicking.rs:391
   6: rust_begin_unwind
             at libstd/panicking.rs:326
   7: core::panicking::panic_fmt
             at libcore/panicking.rs:77
   8: core::panicking::panic_bounds_check
             at libcore/panicking.rs:59
   9: <usize as core::slice::SliceIndex<[T]>>::index_mut
             at /checkout/src/libcore/slice/mod.rs:2052
  10: core::slice::<impl core::ops::index::IndexMut<I> for [T]>::index_mut
             at /checkout/src/libcore/slice/mod.rs:1925
  11: <alloc::vec::Vec<T> as core::ops::index::IndexMut<I>>::index_mut
             at /checkout/src/liballoc/vec.rs:1740
  12: maze::maze::Maze::generate
             at src/maze.rs:166
  13: maze::main
             at src/main.rs:95
  14: std::rt::lang_start::{{closure}}
             at /checkout/src/libstd/rt.rs:74
  15: std::panicking::try::do_call
             at libstd/rt.rs:59
             at libstd/panicking.rs:310
  16: __rust_maybe_catch_panic
             at libpanic_unwind/lib.rs:103
  17: std::rt::lang_start_internal
             at libstd/panicking.rs:289
             at libstd/panic.rs:392
             at libstd/rt.rs:58
  18: std::rt::lang_start
             at /checkout/src/libstd/rt.rs:74
  19: main
  20: __libc_start_main
  21: _start
````

**Comment by tbelaire** [comment]
Fair.  Don't make a maze of size zero :P   It's hard to fit in.

**Comment by tbelaire** [comment]
Fixed in 34764ca4a25e3ae49ba0e92c0b22222b996532d5

**Comment by tbelaire** [comment]
Though if you enter "1 " as a size it still panics, since I use read! instead of trimming and stuff.
